cmd/metal-api/internal/service/vpn-service.go

```diff
@@ -0,0 +1,108 @@
+package service
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	restfulspec "github.com/emicklei/go-restful-openapi/v2"
+
+	"github.com/emicklei/go-restful/v3"
+	"go.uber.org/zap"
+
+	mdmv1 "github.com/metal-stack/masterdata-api/api/v1"
+	mdm "github.com/metal-stack/masterdata-api/pkg/client"
+	"github.com/metal-stack/metal-api/cmd/metal-api/internal/headscale"
+	v1 "github.com/metal-stack/metal-api/cmd/metal-api/internal/service/v1"
+	"github.com/metal-stack/metal-lib/httperrors"
+)
+
+type vpnResource struct {
+	webResource
+	mdc             mdm.Client
+	headscaleClient *headscale.HeadscaleClient
+}
+
+// NewVPN returns a webservice for VPN specific endpoints.
+func NewVPN(
+	log *zap.SugaredLogger,
+	mdc mdm.Client,
+	headscaleClient *headscale.HeadscaleClient,
+) *restful.WebService {
+	r := vpnResource{
+		webResource: webResource{
+			log: log,
+		},
+		mdc:             mdc,
+		headscaleClient: headscaleClient,
+	}
+
+	return r.webService()
+}
+
+// webService creates the webservice endpoint
+func (r *vpnResource) webService() *restful.WebService {
+	ws := new(restful.WebService)
+	ws.
+		Path(BasePath + "v1/vpn").
+		Consumes(restful.MIME_JSON).
+		Produces(restful.MIME_JSON)
+
+	tags := []string{"vpn"}
+
+	ws.Route(ws.GET("/authkey/{pid}").
```

Comment:
`POST` would be a bit better because we will audit these requests and then you could also let the user pass an expiration date in the request payload.